src/flags/blocks.rs

```diff
@@ -97,43 +97,47 @@ impl Blocks {
 
     /// Get a potential `Blocks` struct from a [Config].
     ///
-    /// If the Config's [Yaml] contains an [Array](Yaml::Array) value pointed to by "blocks", each
     /// of its [String](Yaml::String) values is returned in a `Blocks` in a [Some]. Otherwise it
     /// returns [None].
+    /// Config make sure blocks are Strings, we can unwrap here without panic
     fn from_config(config: &Config) -> Option<Self> {
-        if let Some(yaml) = &config.yaml {
-            match &yaml["blocks"] {
-                Yaml::BadValue => None,
-                Yaml::Array(values) => Self::from_yaml_array(values, config),
-                _ => {
-                    config.print_wrong_type_warning("blocks", "array");
-                    None
+        if let Some(c) = &config.blocks {
+            let mut blocks: Vec<Block> = vec![];
+            for b in c.iter() {
+                match Block::try_from(b.as_str().unwrap()) {
+                    Ok(block) => blocks.push(block),
+                    Err(err) => print_error!("bad blocks: {}", err),
```

Comment:
`bad blocks: Not a valid block name: random.` seems a bit old. `Not a valid block name: random.` should be good enough.